mlflow/utils/autologging_utils.py

```diff
@@ -187,3 +192,98 @@ def resolve_input_example_and_signature(
         logger.warning(model_signature_user_msg)
 
     return input_example if log_input_example else None, model_signature
+
+
+# wrapper functions to be able to mock this easily in the tests
+def time_wrapper_for_log():
+    return time.time()
+
+
+def time_wrapper_for_current():
+    return time.time()
+
+
+def time_wrapper_for_timestamp():
+    return time.time()
+
+
+# we pass the batch_metrics_handler through, such that the callback can access it
+def _timed_log_batch(batch_metrics_handler, run_id, metrics):
+    start = time_wrapper_for_log()
+    metrics_slices = [
+        metrics[i * MAX_METRICS_PER_BATCH : (i + 1) * MAX_METRICS_PER_BATCH]
+        for i in range((len(metrics) + MAX_METRICS_PER_BATCH - 1) // MAX_METRICS_PER_BATCH)
+    ]
```

Comment:
I think using the `step` parameter for `range()` will simplify things here. e.g:

```
def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]
```

(Credits to https://stackoverflow.com/a/312464/11952869)